activerecord/lib/active_record/associations/has_one_association.rb

```diff
@@ -87,25 +87,42 @@ def set_new_record(record)
           replace(record, false)
         end
 
-        def remove_target!(method)
+        def remove!(record, method)
```

Comment:
```suggestion
        def remove_record(record, method)
```

Reply:
Shouldn't it be `remove_record!` as the method may raise an exception?

Reply:
No. `!` is only needed if there is a method without the `!`.

Reply:
> `!` is only needed if there is a method without the `!`.

I don't understand this comment. Did you mean "if there is a method **with**" (in other words `!` only if it calls any method with `!`)?

Reply:
`!` should only be used to communicate that it is a "more dangerous" version of a method, so in order to `foo!` exist, `foo` which is the "less dangerous" version, should exist.